### ISSUE TITLE
Indexes in s3 are not deleted

### ISSUE THREAD

**Issue by shokada** [grafana/loki#2944]
**Describe the bug**
Indexes stored in s3 are not deleted with the following error output to the table manager log.
```
level=error ts=2020-11-17T07:15:19.846302Z caller=table_client.go:57 msg="unexpected directories in index_18577 folder, not touching them" directories=[index_18577/]
level=error ts=2020-11-17T07:15:19.867115Z caller=table_client.go:57 msg="unexpected directories in index_18578 folder, not touching them" directories=[index_18578/]
level=error ts=2020-11-17T07:15:19.8829Z   caller=table_client.go:57 msg="unexpected directories in index_18579 folder, not touching them" directories=[index_18579/]
```

**To Reproduce**
Probably occurs when the index store to s3.
My configuration is below. I am using s3 compatible storage.
```
schema_config:
    configs:
        - from: "2020-10-24"
          index:
            period: 24h
            prefix: index_
          object_store: s3
          schema: v11
          store: boltdb-shipper
storage_config:
    aws:
        bucketnames: loki
        endpoint: https://s3-compatible-storage.com
        s3: s3://*:*@https://s3-compatible-storage.com/loki
        s3forcepathstyle: false
    boltdb_shipper:
        active_index_directory: /loki/index
        cache_location: /loki/index_cache
        shared_store: s3
    index_queries_cache_config:
        memcached:
            batch_size: 100
            parallelism: 100
        memcached_client:
            consistent_hash: true
            host: memcached-index-queries.loki.svc.cluster.local
            service: memcached-client
```

**Expected behavior**
The table manager should delete indexes in s3.

**Environment:**
 - Infrastructure: [Kubernetes]
 - Deployment tool: [jsonnet]

**What seems to be the cause**
I think the value of the name variable needs to be changed from `index_18577` to `index_18577/`.
https://github.com/grafana/loki/blob/v2.0.0/pkg/storage/stores/shipper/table_client.go#L51

Without the addition of `/`, `output.Contents` would be empty.
https://github.com/cortexproject/cortex/blob/v1.5.0/pkg/chunk/aws/s3_storage_client.go#L304-L314

**Comment by gcotone** [comment]
I'm also observing this, e.g:
```
msg="table has exceeded the retention period" table=loki_index_18585
msg="deleting table" table=loki_index_18585
msg="unexpected directories in loki_index_18585 folder, not touching them" directories=[loki_index_18585/]

```

**Comment by eatsoup** [comment]
Should be fixed by: #2960 
Thanks to @sandeepsukhani

**Comment by shokada** [comment]
@eatsoup Thank you for letting me know.
I have confirmed that indexes are deleted.
